packages/client-core/src/common/services/ThemeService.tsx

```diff
@@ -299,10 +299,14 @@ export const themes: Record<string, Partial<CSSClasses>> = {
   dark: darkTheme
 }
 
+export const checkDarkMode = () => window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
+
+const initialDarkMode = checkDarkMode()
+
 export const ThemeState = defineState({
   name: 'ThemeState',
   initial: {
-    theme: 'dark' as 'light' | 'dark'
+    theme: (initialDarkMode ? 'dark' : 'light') as 'light' | 'dark'
```

Comment:
prefer an initializer function rather than top level declaration:

```ts

export const ThemeState = defineState({
  name: 'ThemeState',
  initial: () => ({
    theme: (initialDarkMode ? 'dark' : 'light') as 'light' | 'dark'
  }))
})

Reply:
sorry i mean that we should call checkDarkMode such that we aren't accessing the window at module scope, as it is this will make it hard to write tests

Reply:
Ah I figured you meant something like that